instance/models/mixins/storage.py

```diff
@@ -255,7 +255,12 @@ def s3_custom_domain(self):
         """
         The custom domain name built based on the bucket name.
         """
-        return "{}.s3.amazonaws.com".format(self.s3_bucket_name)
+        # If s3_region is empty, boto3 will use S3 default region 'N. Virginia (us-east-1)'
+        # Reference: https://docs.aws.amazon.com/general/latest/gr/rande.html
+        return "{}.s3.{}.amazonaws.com".format(
+            self.s3_bucket_name,
+            self.s3_region or 'us-east-1'
```

Comment:
I'm glad this works, `boto3` does not accept explicit `us-east-1` in many places.

Reply:
This is a hacky way to construct a direct url from the bucket (see [here](https://django-storages.readthedocs.io/en/latest/backends/amazon-S3.html), `AWS_S3_CUSTOM_DOMAIN`). Probably used because some weird behaviour in the platform.